service/src/main/java/bio/terra/pipelines/service/PipelinesService.java

```diff
@@ -41,14 +47,116 @@ public Pipeline getPipeline(PipelinesEnum pipelineName) {
    * This method is meant to only be called by an admin endpoint to update the control workspace id
    * for a pipeline
    *
-   * @param pipelineName - nanme of pipeline to update
+   * @param pipelineName - name of pipeline to update
    * @param workspaceId - UUID of workspace to update to
-   * @return
+   * @return pipeline with updated workspaceId
    */
   public Pipeline updatePipelineWorkspaceId(PipelinesEnum pipelineName, UUID workspaceId) {
     Pipeline pipeline = getPipeline(pipelineName);
     pipeline.setWorkspaceId(workspaceId);
     pipelinesRepository.save(pipeline);
     return pipeline;
   }
+
+  public void validateInputs(PipelinesEnum pipelineName, Object inputs) {
+    Pipeline pipeline = getPipeline(pipelineName);
+    List<PipelineInputDefinition> inputDefinitions = pipeline.getPipelineInputDefinitions();
+
+    LinkedHashMap<String, Object> inputsMap = castInputsToMap(inputs);
+
+    ArrayList<String> errorMessages =
+        new ArrayList<>(validateRequiredInputs(inputDefinitions, inputsMap));
+
+    errorMessages.addAll(validateInputTypes(inputDefinitions, inputsMap));
+
+    checkForExtraInputs(pipeline, inputDefinitions, inputsMap);
+
+    if (!errorMessages.isEmpty()) {
+      throw new ValidationException(
+          String.format("Problem(s) with pipelineInputs: %s", String.join("; ", errorMessages)));
+    }
+  }
+
+  private LinkedHashMap<String, Object> castInputsToMap(Object inputs) {
+    try {
+      return objectMapper.convertValue(inputs, LinkedHashMap.class);
```

Comment:
Does this need to explicitly return a `LinkedHashMap`?  Could it just return a Map?

Also, to preserve types (in case you get a warning) you do this:
```suggestion
      return objectMapper.convertValue(inputs, new TypeReference<LinkedHashMap<String, Object>(){});
```